### PR TITLE
Persistence Component: Nontemporal Delta + Bulk Load for DuckDB

### PR DESCRIPTION
**Pull request by kumuwu**
#### What type of PR is this?

- Improvement

#### What does this PR do / why is it needed ?

- Nontemporal Delta was unnecessarily updating the primary keys (redundant updates). Removed such updates and with this we can enable this mode for DuckDB.
- Added basic usage of Bulk Load without digest generation

#### Which issue(s) this PR fixes:

Fixes #

### CODE REVIEW

File: legend-engine-xts-persistence/legend-engine-xt-persistence-component/legend-engine-xt-persistence-component-relational-duckdb/src/main/java/org/finos/legend/engine/persistence/components/relational/duckdb/sqldom/schemaops/expressions/table/FileRead.java [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+public class FileRead extends TableLike
+{
+    public enum FileType
+    {
+        CSV,
+        JSON,
+        PARQUET
+    }
+
+    private final FileType fileType;
```
Comment (prasar-ashutosh) [inline legend-engine-xts-persistence/legend-engine-xt-persistence-component/legend-engine-xt-persistence-component-relational-duckdb/src/main/java/org/finos/legend/engine/persistence/components/relational/duckdb/sqldom/schemaops/expressions/table/FileRead.java:47]: Let's add a validate method to test what all is mandatory.

File: legend-engine-xts-persistence/legend-engine-xt-persistence-component/legend-engine-xt-persistence-component-relational-duckdb/src/main/java/org/finos/legend/engine/persistence/components/relational/duckdb/sqldom/schemaops/statements/CopyStatement.java [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+    private Table table;
+    private final List<Field> columns;
+    private SelectStatement selectStatement;
+
+    public CopyStatement()
+    {
+        columns = new ArrayList<>();
+    }
+
+    public CopyStatement(Table table, List<Field> columns, SelectStatement selectStatement)
```
Comment (prasar-ashutosh) [inline legend-engine-xts-persistence/legend-engine-xt-persistence-component/legend-engine-xt-persistence-component-relational-duckdb/src/main/java/org/finos/legend/engine/persistence/components/relational/duckdb/sqldom/schemaops/statements/CopyStatement.java:43]: Let's write a dedicated test for this

File: legend-engine-xts-persistence/legend-engine-xt-persistence-component/legend-engine-xt-persistence-component-relational-duckdb/src/main/java/org/finos/legend/engine/persistence/components/relational/duckdb/sqldom/schemaops/expressions/table/FileRead.java (hunk 1 above)
Comment (kumuwu) [reply legend-engine-xts-persistence/legend-engine-xt-persistence-component/legend-engine-xt-persistence-component-relational-duckdb/src/main/java/org/finos/legend/engine/persistence/components/relational/duckdb/sqldom/schemaops/expressions/table/FileRead.java:47]: Sure added!

File: legend-engine-xts-persistence/legend-engine-xt-persistence-component/legend-engine-xt-persistence-component-relational-duckdb/src/main/java/org/finos/legend/engine/persistence/components/relational/duckdb/sqldom/schemaops/statements/CopyStatement.java (hunk 2 above)
Comment (kumuwu) [reply legend-engine-xts-persistence/legend-engine-xt-persistence-component/legend-engine-xt-persistence-component-relational-duckdb/src/main/java/org/finos/legend/engine/persistence/components/relational/duckdb/sqldom/schemaops/statements/CopyStatement.java:43]: Sure added!

**Review by epsstan — APPROVED** [review]

**Pull request merged**